lib/routes/tsdm39/bd.ts

```diff
@@ -0,0 +1,100 @@
+import type { DataItem, Route } from '@/types';
+import { config } from '@/config';
+import ofetch from '@/utils/ofetch';
+import { load } from 'cheerio';
+import cache from '@/utils/cache';
+import { parseDate } from '@/utils/parse-date';
+
+// type id => display name
+type Mapping = Record<string, string>;
+
+const TYPE: Mapping = {
+    '403': '720P',
+    '404': '1080P',
+    '405': 'BDMV',
+    '4130': '4K',
+    '5815': 'AV1',
+};
+
+// render into MD table
+const mkTable = (mapping: Mapping): string => {
+    const heading: string[] = [],
+        separator: string[] = [],
+        body: string[] = [];
+
+    for (const key in mapping) {
+        heading.push(mapping[key]);
+        separator.push(':--:');
+        body.push(key);
+    }
+
+    return [heading.join(' | '), separator.join(' | '), body.join(' | ')].map((s) => `| ${s} |`).join('\n');
+};
+
+const handler: Route['handler'] = async (ctx) => {
+    const { type } = ctx.req.param();
+
+    const cookie = config.tsdm39.cookie;
```

Comment:
Please throw an exception if cookie is not set.

Reply:
From my understanding from the description, I think this route will work without cookie if models are SFW.